osquery/sql/virtual_table.cpp

```diff
@@ -126,20 +126,24 @@ int xColumn(sqlite3_vtab_cursor *cur, sqlite3_context *ctx, int col) {
   if (type == "TEXT") {
     sqlite3_result_text(ctx, value.c_str(), value.size(), SQLITE_STATIC);
   } else if (type == "INTEGER") {
+    char *end;
     int afinite;
-    try {
-      afinite = boost::lexical_cast<int>(value);
-    } catch (const boost::bad_lexical_cast &e) {
+    long int val = strtol(value.c_str(), &end, 10);
+    if (end == value.c_str() || *end != '\0' ||
```

Comment:
Should this first include a check that `end != nullptr` before the dereference check to `\0`?